### PR TITLE
E2E test for scheduled Draft feature

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
zephyr test cases PR: https://github.com/mattermost/mattermost-test-management/pull/183

Fixes https://mattermost.atlassian.net/browse/MM-60374

```release-note
NONE
```

### CODE REVIEW

File: e2e-tests/playwright/support/ui/components/channels/scheduled_draft_modal.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    readonly dateLocator;
+
+    constructor(container: Locator) {
+        this.container = container;
+
+        this.confirmButton = container.locator('button.confirm');
+        this.dateInput = container.locator('div.Input_wrapper');
+        this.timeLocator = container.locator('div.dateTime__input');
+        this.timeDropdownOptions = container.locator('ul.dropdown-menu .MenuItem');
+        this.dateLocator = (day: number, month: string) => container.locator(`button[aria-label*='${day}th ${month}']`);
```
Comment (saturninoabril) [inline e2e-tests/playwright/support/ui/components/channels/scheduled_draft_modal.ts:22]: Maybe the date format here is not accurate? In some case, I'm seeing like `1st November (Friday)`

File: e2e-tests/playwright/support/ui/components/channels/post_create.ts [hunk 2]
```diff
@@ -5,12 +5,12 @@ import {expect, Locator} from '@playwright/test';
 
 export default class ChannelsPostCreate {
     readonly container: Locator;
-
     readonly input;
 
     readonly attachmentButton;
     readonly emojiButton;
     readonly sendMessageButton;
+    readonly scheduleDraftMessageButton;
```
Comment (saturninoabril) [inline e2e-tests/playwright/support/ui/components/channels/post_create.ts:13]: I'm seeing different variants such as:
```
scheduleMessage...
scheduledMessage...
scheduleDraftMessage...
scheduledDarft...
```

I wonder if this should follow a common prefix only for consistency and easier to reuse/remember.

File: e2e-tests/playwright/tests/functional/channels/scheduled_drafts/create_scheduled_draft.spec.ts [hunk 3]
```diff
@@ -0,0 +1,222 @@
+import {expect, Page} from '@playwright/test';
+import {test} from '@e2e-support/test_fixture';
+import {duration, wait} from '@e2e-support/util';
+import {ChannelsPage, ScheduledMessagePage} from '@e2e-support/ui/pages';
+
+test('Should create a scheduled message from a channel', async ({pw, pages}) => {
```
Comment (saturninoabril) [inline e2e-tests/playwright/tests/functional/channels/scheduled_drafts/create_scheduled_draft.spec.ts:6]: Once available, please update this test and the next one with corresponding test IDs.

File: webapp/channels/src/components/custom_status/date_time_input.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ import {relativeFormatDate} from 'utils/datetime';
 import {isKeyPressed} from 'utils/keyboard';
 import {getCurrentMomentForTimezone} from 'utils/timezone';
 
-const CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES = 30;
+const CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES = 1;
```
Comment (saturninoabril) [inline webapp/channels/src/components/custom_status/date_time_input.tsx:31]: It would be great to make this configurable so this can be made testable.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks Yasser! Great to see to have these written in Playwright!

Overall looks good except for few suggestions. Lastly, please ensure that it passed all Playwright projects (ipad, chrome and firefox). Currently, I've seen it failed in firefox.
Screenshot 2024-10-18 at 9 49 27 AM

File: e2e-tests/playwright/support/ui/components/channels/post_create.ts (hunk 2 above)
Comment (yasserfaraazkhan) [reply e2e-tests/playwright/support/ui/components/channels/post_create.ts:13]: Got it. I'll use scheduledDraft overall.

File: e2e-tests/playwright/support/ui/components/channels/scheduled_draft_modal.ts (hunk 1 above)
Comment (yasserfaraazkhan) [reply e2e-tests/playwright/support/ui/components/channels/scheduled_draft_modal.ts:22]: Updated the locator to consider `st | nd | rd | th` for the date

File: e2e-tests/playwright/tests/functional/channels/scheduled_drafts/create_scheduled_draft.spec.ts (hunk 3 above)
Comment (harshilsharma63) [reply e2e-tests/playwright/tests/functional/channels/scheduled_drafts/create_scheduled_draft.spec.ts]: Done. This PR - https://github.com/mattermost/mattermost/pull/28918 does vi via dev mode.

File: e2e-tests/playwright/tests/functional/channels/scheduled_drafts/create_scheduled_draft.spec.ts [hunk 5]
```diff
@@ -0,0 +1,416 @@
+import {expect, Page} from '@playwright/test';
+import {test} from '@e2e-support/test_fixture';
+import {duration, wait} from '@e2e-support/util';
+import {ChannelsPage, ScheduledDraftPage} from '@e2e-support/ui/pages';
+
+test('should create a scheduled message from a channel', async ({pw, pages}) => {
```
Comment (saturninoabril) [inline e2e-tests/playwright/tests/functional/channels/scheduled_drafts/create_scheduled_draft.spec.ts:6]: Please add the test case ID here.

**Review by saturninoabril — APPROVED** [review]
Thanks @yasserfaraazkhan, looks good to me except for the request to add the test case IDs. Approving now given it will be added.

**Comment by agnivade** [comment]
I think the schema migrations here are a merge conflict issue. If so, please remove server-platform from reviewers. Thanks.

**Pull request merged**